script.js

```diff
@@ -168,42 +188,82 @@ Chart.prototype = {
     var chart = this;
 
     // TRANSFORM DATA
+    txData = app.data.slice();
+
+    if (app.options.filtered) {
+        txData = txData.filter(function (d) { return d.level === app.options.filtered; });
+    } 
+
+    if (app.options.socgroup !== 'all') {
+      var socgroup=app.options.socgroup;
+      txData = txData.filter(function (d) {
+        return d.grp_text === socgroup;
+      });
+    }
+
+    d3.selectAll('#soc-group').classed('disabled', app.options.filtered=='agg');
 
-    var txData = app.data.filter(function (d) { return d.year === app.options.year; });
 
     var t = d3.transition().duration(TRANSITION_DURATION)
 
     // UPDATE CHART ELEMENTS
 
-    var yearText = d3.selectAll('.year')
-      .transition().delay(TRANSITION_DURATION / 2)
-      .text(app.options.year);
-
       //return d.country is a key, look for circle with that label year after year
       //data function takes data then the key
-    var countries = chart.svg.selectAll('.country')
-      .data(txData, function (d) { return d.country; });
+    var points=chart.svg.selectAll('.point')
+      .data(txData, function (d) { return d.socname; });
 
       //.merge and after is created and previously existing circles
       //before .merge only applies to the new selections
-    countries.enter().append('circle')
-      .attr('class', 'country')
-      .style('fill', function (d) { return chart.color(d.continent); })
-      .style('opacity', 0.75)
+    points.enter().append('circle')
+      .attr('class','point')
       .attr('r', 0)
-      .attr('cx', chart.width / 2)
-      .attr('cy', chart.height / 2)
-      .merge(countries)
-      .sort(function (a, b) { return b.population - a.population; })
+      .attr('cx', function(d) { return chart.x(d.femper); })
+      .attr('cy',  function(d) { return chart.y(d.earn); })
+      .style('stroke',  function (d) {
+        var returnColor;
+        if (d.wagegap_group===1) {
+              returnColor='#0A6E2c';
+            } else if (d.wagegap_group===2) {
+              returnColor='#968A0D';
+            } else if (d.wagegap_group===3) {
+              returnColor='#B15500';
+            } else if (d.wagegap_group===4) {
+              returnColor='#96230D';
+            } else returnColor='#96230D';
+            return returnColor;
```

Comment:
D3 has a scale for this: the [threshold scale](https://github.com/d3/d3-scale/blob/master/README.md#threshold-scales).

Might there be a better color range to use here? This one has some particular problems for users with color blindness. Maybe [ColorBrewer](http://colorbrewer2.org) has some options?